### PR TITLE
Introduce Access Card management

### PR DESCRIPTION
**Pull request by lukaszkarykowski**

### CODE REVIEW

File: src/ralph/access_cards/models.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        null=True,
+        blank=True,
+        help_text=_('Date of issue to the User')
+    )
+    notes = models.TextField(
+        null=True,
+        blank=True,
+        help_text=_('Optional notes')
+    )
+    user = models.ForeignKey(
```
Comment (ar4s) [inline src/ralph/access_cards/models.py:50]: Please add ``on_delete`` attribute. I recommend you ``on_delete=models.SET_NULL`` in this case.

File: src/ralph/access_cards/models.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        RalphUser,
+        null=True,
+        blank=True,
+        related_name='used_access_cards',
```
Comment (ar4s) [inline src/ralph/access_cards/models.py]: I don't see any usage of backward relation. Please consider ``+`` if you don't need.

File: src/ralph/access_cards/models.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        help_text=_('User of the card')
+    )
+    owner = models.ForeignKey(
+        RalphUser,
+        null=True,
+        blank=True,
+        related_name='owned_access_cards',
```
Comment (ar4s) [inline src/ralph/access_cards/models.py]: ditto

**Review by ar4s — CHANGES_REQUESTED** [review]

File: src/ralph/access_cards/models.py (hunk 1 above)
Comment (ar4s) [reply src/ralph/access_cards/models.py:50]: Are there any limits? Can every user have an unlimited number of cards?

**Review by ar4s — CHANGES_REQUESTED** [review]

File: src/ralph/access_cards/models.py (hunk 2 above)
Comment (lukaszkarykowski) [reply src/ralph/access_cards/models.py]: changed.

File: src/ralph/access_cards/models.py (hunk 1 above)
Comment (lukaszkarykowski) [reply src/ralph/access_cards/models.py:50]: Added.

Comment (lukaszkarykowski) [reply src/ralph/access_cards/models.py:50]: Yes, User can have any number of cards.

File: src/ralph/access_cards/models.py (hunk 3 above)
Comment (lukaszkarykowski) [reply src/ralph/access_cards/models.py]: Changed.

**Review by ar4s — DISMISSED** [review]

File: src/ralph/access_cards/models.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    status = ChoiceField(
+        choices=AccessCardStatus,
+        default=AccessCardStatus.new.id,
+        null=False,
+        blank=False,
+        help_text=_('Access card status')
+    )
+
+    def __str__(self):
+        return _('Access Card: {}').format(self.visual_number)
```
Comment (romcheg) [inline src/ralph/access_cards/models.py:75]: It's better override the default autocomplete queryset in order to filter out liquidated cards.

```    @classmethod
    def get_autocomplete_queryset(cls):
        return cls._default_manager.exclude(
            status=AccessCardStatus.liquidated.id
        )```

File: src/ralph/access_cards/tests/test_api.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            access_card1, response.data['results'][1]
+        )
+
+    def test_class_access_card_test_case(self):
+        access_card = {
+            'status':"in use",
+            'visual_number':'654321',
+            'system_number':'F9876DSGV',
+            'notes':'test note',
+            'issue_date':'2020-01-02'
```
Comment (romcheg) [inline src/ralph/access_cards/tests/test_api.py]: `region` is a required field

File: src/ralph/access_cards/tests/factories.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import factory
+from factory.django import DjangoModelFactory
+
+from ralph.access_cards.models import AccessCard, AccessCardStatus
+
+
+class AccessCardFactory(DjangoModelFactory):
+    visual_number = factory.Sequence(lambda n: '000000000000{}'.format(n))
+    system_number = factory.Sequence(lambda n: '000000000000{}'.format(n))
+    status = AccessCardStatus.new
```
Comment (romcheg) [inline src/ralph/access_cards/tests/factories.py:12]: region is a required field

File: src/ralph/access_cards/tests/test_api.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            access_card.modified.isoformat()
+        )
+        self.assertEqual(response_data['notes'], access_card.notes)
+        self.assertEqual(
+            response_data['user']['username'],
+            access_card.user.username
+        )
+        self.assertEqual(
+            response_data['owner']['username'],
+            access_card.owner.username
```
Comment (romcheg) [inline src/ralph/access_cards/tests/test_api.py:36]: Need to check the region as well.

**Review by romcheg — CHANGES_REQUESTED** [review]

File: src/ralph/access_cards/models.py (hunk 4 above)
Comment (lukaszkarykowski) [reply src/ralph/access_cards/models.py:75]: Added.

File: src/ralph/access_cards/tests/test_api.py (hunk 5 above)
Comment (lukaszkarykowski) [reply src/ralph/access_cards/tests/test_api.py]: Added

File: src/ralph/access_cards/tests/test_api.py (hunk 7 above)
Comment (lukaszkarykowski) [reply src/ralph/access_cards/tests/test_api.py:36]: Added.

File: src/ralph/access_cards/tests/factories.py (hunk 6 above)
Comment (lukaszkarykowski) [reply src/ralph/access_cards/tests/factories.py:12]: Added.

**Review by ar4s — APPROVED** [review]

**Review by romcheg — APPROVED** [review]
LGTM

**Pull request merged**